src/lex.ll

```diff
@@ -512,20 +512,42 @@ L?\"(\\.|[^\\"])*\" { lStringConst(&yylval, &yylloc); return TOKEN_STRING_LITERA
 {FLOAT_NUMBER} {
     RT;
     std::string val(yytext);
+    std::string fp64S("d");
+    std::string fp64C("D");
     std::string fp16S("f16");
     std::string fp16C("F16");
     if (val.size() >= fp16S.size() && ((val.compare(val.size() - fp16S.size(), fp16S.size(), fp16S) == 0)
            || (val.compare(val.size() - fp16C.size(), fp16C.size(), fp16C) == 0))) {
         yylval.stringVal = new std::string(val.substr(0, val.length() - 3));
         return TOKEN_FLOAT16_CONSTANT;
+    } else if (val.size() >= fp64S.size() && ((val.compare(val.size() - fp64S.size(), fp64S.size(), fp64S) == 0)
+           || (val.compare(val.size() - fp64C.size(), fp64C.size(), fp64C) == 0))) {
+        yylval.doubleVal = atof(yytext);
```

Comment:
Any reason to do that through `atof` now, when we use APFloat to parse it on other code paths?

Also `(float)atof(yytext)` is bad practice as it parses as double and converts to float.